Teleportation/ReferenceImplementation.qs

```diff
@@ -147,6 +147,10 @@ namespace Quantum.Kata.Teleportation {
     operation EntangleThreeQubits_Reference (qAlice : Qubit, qBob : Qubit, qCharlie : Qubit) : Unit
     is Adj {
         
+        // |000⟩ --H⊗H⊗I--> 1/2(|000⟩+|010⟩+|100⟩+|110⟩) --X(c)-> 1/2(|001⟩+|011⟩+|101⟩+|111⟩)
```

Comment:
I think this explanation would be a bit more readable if the states were given in comments between the actual gates. This way the learner can also add `DumpMachine` between the gates and verify that the output matches the comment. So the start of the code would look like

    // |000⟩ 

    H(qAlice);

    // 1/sqrt(2) (|0⟩ + |1⟩)|00⟩
    
    H(qBob);
    
    // 1/2 (|000⟩+|010⟩+|100⟩+|110⟩)

and so on.